src/main/java/com/bnkrll/service/SessionRepositorySimulator.java

```diff
@@ -0,0 +1,49 @@
+package com.bnkrll.service;
+
+import com.bnkrll.model.Session;
+import com.fasterxml.jackson.annotation.ObjectIdGenerators;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Repository;
+
+import java.util.*;
+
+@Slf4j
+@Repository
+public class SessionRepositorySimulator implements SessionRepository{
+
+    private final Map<String, Session> data;
+
+    private static final String SESSION_ID_ONE = "bb791404-78f0-11ed-a1eb-0242ac120002";
+    private static final String SESSION_ID_TWO = "ada8881e-78f0-11ed-a1eb-0242ac120002";
+
+    public SessionRepositorySimulator() {
+        this.data = new HashMap<>();
+    }
+
+
+    @Override
+    public Session findById(String sessionID) {
+        return this.data.get(sessionID);
+    }
+
+    @Override
+    public void save(Session session) {
+        this.data.put(session.getSessionId(),session);
+        log.info("Session with id {} saved!", session.getSessionId());
+    }
+
+    @Override
+    public List<Session> getLastSessions(int numofSessions) {
+        for(int i = 0; i < numofSessions; i++){
+            this.data.put(createSessions().getSessionId(),createSessions());
```

Comment:
Instead of setting up a stub you should actually code out how you would retrieve the last x values from this hashmap. I can help you, but I'd like to see you give it a try first.

Reply:
Any hints, please?

Reply:
Made a change at how to retrieve the last x values here - https://github.com/logunlaja26/BNKRLL-SERVICE/commit/c215c786d7fb1a458ce81c549b84459432b98d28